### PR TITLE
Add Leafcutter tool

### PR DESCRIPTION
**Pull request by lpantano**
Add first steps of leaf cutter splicing quantification to the pipeline.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnasplice/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/rnasplice _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -392,6 +392,35 @@ if (params.source == "fastq" | params.source == "genome_bam") {
...
 //
 
+if (params.source == "fastq" | params.source == "genome_bam") {
+
+    if (params.leafcutter) {
+
+        process {
+
+            withName: 'REGTOOLS_JUNCTIONSEXTRACT' {
+                ext.args   = '-a 8 -i 50 -I 500000'
```
Comment (mashehu) [inline conf/modules.config:405]: are there long names for these arguments available? If so, please use them instead to make it more readable.

File: docs/output.md [hunk 2]
```diff
@@ -295,6 +296,18 @@ This workflow will also produce a `suppa_tpm.txt` which is a tab delimitated tex
...
+### Leafcutter
+
+<details markdown="1">
+<summary>Output files</summary>
+
+- `leafcutter`
+  - `*junc`: all junctions found in each sample and their abundances.
+  - `*numers.counts.gz`: Each column corresponds to a different sample (original bam file) and each row to an intron, which are identified as chromosome:intron_start:intron_end:cluster_id. This is the file to use for downstream analysis.
+  - `*counts.gz`: Same than before but with the ratio of that event.
+
```
Comment (mashehu) [inline docs/output.md:310]: any explanation about the tool, including link to its homepage?

File: nextflow_schema.json [hunk 3]
```diff
@@ -437,7 +437,7 @@
             "properties": {
                 "aligner": {
                     "type": "string",
-                    "default": "star",
+                    "default": "star_salmon",
                     "description": "Specifies the alignment algorithm to use - available options are 'star_salmon', or 'star'.",
```
Comment (mashehu) [inline nextflow_schema.json:442]: please add an `enum` entry for this parameter to improve validation and documentation

File: nextflow_schema.json [hunk 4]
```diff
@@ -654,7 +666,7 @@
                 },
                 "miso_genes": {
                     "type": "string",
-                    "default": "ENSG00000004961, ENSG00000005302",
+                    "default": "ENSG00000004961, ENSG00000005302, ENSG00000147403",
                     "description": "List containing identifiers of genes to plot."
```
Comment (mashehu) [inline nextflow_schema.json]: please specify the format of the list, and add a pattern entry to avoid csv/tsv confusion etc.

**Review by mashehu — CHANGES_REQUESTED** [review]

File: nextflow_schema.json (hunk 4 above)
Comment (lpantano) [reply nextflow_schema.json]: can I get an example of the pattern?

File: conf/modules.config (hunk 1 above)
Comment (lpantano) [reply conf/modules.config:405]: no long parameters for this. I will add some comments

File: nextflow_schema.json (hunk 4 above)
Comment (mashehu) [reply nextflow_schema.json]: something like
```suggestion
                    "description": "List containing identifiers of genes to plot."
                    "pattern": "^[\\w-]+(,\\s+[\\w-]+)*$"
```

**Comment by mashehu** [comment]
fixed the linting action. just guessed what the current template version for this pipeline code is. please change the version in `.nf-core.yml` to the actual version, so the linting is a bit more precise.

**Review by mashehu — APPROVED** [review]

**Comment by nvnieuwk** [comment]
I'll merge this, the linting will be fixed in another PR

**Pull request merged**